src/ruby/ext/grpc/extconf.rb

```diff
@@ -24,10 +24,18 @@
 
 ENV['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
 
-ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
-ENV['CC'] = RbConfig::CONFIG['CC']
-ENV['CXX'] = RbConfig::CONFIG['CXX']
-ENV['LD'] = ENV['CC']
+if ENV['AR'].nil? || ENV['AR'].size == 0
+    ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
+end
+if ENV['CC'].nil? || ENV['CC'].size == 0
+    ENV['CC'] = RbConfig::CONFIG['CC']
+end
+if ENV['CXX'].nil? || ENV['CXX'].size == 0
+    ENV['CXX'] = RbConfig::CONFIG['CXX']
+end
+if ENV['LD'].nil? || ENV['LD'].size == 0
+    ENV['LD'] = ENV['LD']
```

Comment:
@matoro I'm sorry to go back and forth on this, but I need to ask for this change back - it was right before to have `ENV['LD'] = ENV['CC']`. Chatted with @nicolasnoble and it makes sense to delegate the linker to `CC` in this context

Reply:
Okay.  I can imagine an environment where a linker override would be necessary (gold vs PE linker?) but don't think it's really such a huge deal either way.